test/unit/math/rev/core/save_varis_test.cpp

```diff
@@ -0,0 +1,414 @@
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+using stan::math::var;
+using stan::math::vari;
+
+TEST(AgradRev_save_varis, int_arg) {
```

Comment:
I think the recursion ending function is not tested for save_varis. Can u add that please?